app/components/post_list/post_list.js

```diff
@@ -24,37 +24,74 @@ const style = StyleSheet.create({
     }
 });
 
-export default class PostList extends React.Component {
+const LOAD_MORE_POSTS = 'load-more-posts';
+
+export default class PostList extends Component {
     static propTypes = {
-        posts: React.PropTypes.array.isRequired,
-        theme: React.PropTypes.object.isRequired,
-        onPostPress: React.PropTypes.func,
-        renderReplies: React.PropTypes.bool,
-        indicateNewMessages: React.PropTypes.bool,
-        currentUserId: React.PropTypes.string,
-        lastViewedAt: React.PropTypes.number
+        posts: PropTypes.array.isRequired,
+        postsRequests: PropTypes.object.isRequired,
+        theme: PropTypes.object.isRequired,
+        allowLoadMore: PropTypes.bool,
+        loadMore: PropTypes.func,
+        onPostPress: PropTypes.func,
+        renderReplies: PropTypes.bool,
+        indicateNewMessages: PropTypes.bool,
+        currentUserId: PropTypes.string,
+        lastViewedAt: PropTypes.number
     };
 
     constructor(props) {
         super(props);
-        const {posts, indicateNewMessages, currentUserId, lastViewedAt} = this.props;
         this.state = {
+            hasFirstPost: false,
+            didInitialPostsLoad: false,
             dataSource: new ListView.DataSource({
                 rowHasChanged: (a, b) => a !== b
-            }).cloneWithRows(addDatesToPostList(posts, indicateNewMessages, currentUserId, lastViewedAt))
+            }).cloneWithRows(this.getPostsWithDates(props))
         };
     }
 
     componentWillReceiveProps(nextProps) {
+        const didInitialPostsLoad = this.didPostsLoad(nextProps, 'getPosts') ||
```

Comment:
@hmhealey I totally agree that the logic for loading more posts belongs in the `ChannelPostList`, but unfortunately the button/indicator for loading more posts must be rendered in the `ListView` of the `PostList` (for layout/style reasons). This is why I initially put the logic here. I intended to move it to the parent component since the beginning of the PR, but I eventually forgot to do that. Thanks for the reminder.

At another point we can figure out the layout challenges with rendering the `LoadMorePosts` inside of the `ChannelPostList`.